server/inspect.go

```diff
@@ -0,0 +1,51 @@
+package server
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+
+	"github.com/go-zoo/bone"
+)
+
+type ContainerInfo struct {
+	Pid         int               `json:"pid"`
+	Image       string            `json:"image"`
+	CreatedTime int64             `json:"created_time"`
+	Labels      map[string]string `json:"labels"`
+}
+
+func (s *Server) StartInspectEndpoint() error {
+	mux := bone.New()
+
+	mux.Get("/containers/:id", http.HandlerFunc(func(w http.ResponseWriter, req *http.Request) {
+		containerId := bone.GetValue(req, "id")
+		ctr := s.GetContainer(containerId)
+		if ctr == nil {
+			http.Error(w, fmt.Sprintf("container with id: %s not found", containerId), http.StatusNotFound)
+			return
+		}
+		ctrState := ctr.State()
+		if ctrState == nil {
+			http.Error(w, fmt.Sprintf("container %s state is nil", containerId), http.StatusNotFound)
+			return
+		}
+
+		ci := ContainerInfo{
+			Pid:         ctrState.Pid,
+			Image:       ctr.Image(),
+			CreatedTime: ctrState.Created.UnixNano(),
+			Labels:      ctr.Labels(),
+		}
```

Comment:
we do need annotations https://github.com/derekwaynecarr/kubernetes/pull/4/files#diff-84f52d75d93d7dd10d29024469937e2eR211 or get `restartCount` from container itself

Reply:
Added annotations.